### PR TITLE
chore: organize sidebar a bit

### PR DESCRIPTION
**Pull request by aramissennyeydd**
## Hey, I just made a Pull Request!

From the docs SIG - proposing a small change in side bar organization to reduce sprawl. Primary goal here is to leverage the "Core Features" and "References" sections more, I moved permissions, auth and notifications to core features and then shifted the ADRs and API reference to the references section.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Parsifal-M — COMMENTED** [review]
Nice! I think it makes sense! I am all for making things more organised 🤓

File: microsite/sidebars.js [hunk 1]
```diff
@@ -496,30 +505,54 @@ module.exports = {
...
+            label: 'Guides',
+            items: ['api/utility-apis'],
+          },
+          {
+            type: 'category',
+            label: 'API Reference',
+            items: ['reference/index'],
+          },
+          'api/deprecations',
+        ],
```
Comment (Rugvip) [inline microsite/sidebars.js:533]: I think we can flatten these out into the parent, i.e. all of these directly under "References"

File: microsite/sidebars.js [hunk 2]
```diff
@@ -116,6 +116,66 @@ module.exports = {
           'features/search/how-to-guides',
         ],
       },
+      {
+        type: 'category',
+        label: 'Auth and Identity',
```
Comment (Rugvip) [inline microsite/sidebars.js:121]: Let's sort all entries in this section alphabetically

File: microsite/sidebars.js [hunk 3]
```diff
@@ -276,59 +341,6 @@ module.exports = {
...
-        items: [
-          'permissions/plugin-authors/01-setup',
-          'permissions/plugin-authors/02-adding-a-basic-permission-check',
-          'permissions/plugin-authors/03-adding-a-resource-permission-check',
-          'permissions/plugin-authors/04-authorizing-access-to-paginated-data',
-          'permissions/plugin-authors/05-frontend-authorization',
-        ],
-      },
-    ],
     Tooling: [
```
Comment (Rugvip) [inline microsite/sidebars.js:344]: Discussed a bit in the SIG and figured it's probably cleanest to flatten all of the tooling sections under a single "Backstage CLI" section, and then put that under "Framework". For example we'd have "Framework/Backstage CLI/Linking in Local Packages"

**Review by Rugvip — COMMENTED** [review]
Nice! 🎉 

Discussed this in the docs SIG and came up with a few tweaks before :shipit:

File: microsite/sidebars.js (hunk 3 above)
Comment (aramissennyeydd) [reply microsite/sidebars.js]: updated in [0d6efa4](https://github.com/backstage/backstage/pull/27803/commits/0d6efa49ac7931a721e7fda97766947feead8c62)

File: microsite/sidebars.js (hunk 2 above)
Comment (aramissennyeydd) [reply microsite/sidebars.js:54]: updated in [0d6efa4](https://github.com/backstage/backstage/pull/27803/commits/0d6efa49ac7931a721e7fda97766947feead8c62)

File: microsite/sidebars.js (hunk 1 above)
Comment (aramissennyeydd) [reply microsite/sidebars.js]: Updated as well, I think some of these have a better home elsewhere - utility APIs seems related to the (old?) frontend system and deprecations has 2 entries from years ago. Going to keep them here, but worth coming back to when we want to do a more in depth rewrite.

**Review by Rugvip — APPROVED** [review]
Nice, thank you! 👍 

Let's :shipit: 🎉

**Pull request merged**